requirements.txt

```diff
@@ -8,11 +8,18 @@ googlemaps~=4.5.3
 pandas-gbq~=0.19.1
 pydata-google-auth~=1.7.0
 pandera[io]~=0.8.1
-jupyterlab~=3.2.8
+jupyterlab~=4.3.0
 pandas~=1.4.0
 joblib~=1.2.0
 beautifulsoup4~=4.11
 plotly~=5.15.0
 gridstatus~=0.20.0
 s3fs>=2022.11.0
 click>=8.1.7
+numpy<2
+sqlalchemy<2
+coloredlogs~=15.0.1
+scipy~=1.14.1
+google-cloud-storage~=2.18.2
+geopandas~=1.0.1
+gcsfs~=2024.10.0
```

Comment:
Some of these dependencies were coming from PUDL so I made them direct dependencies.